libxl: Add -xen-acpi-pm for acpi support

This enables OpenXT's enhanced ACPI support for the VM.  Notably, this
provides the hvm-powerbutton-enable xenstore entry for the stubdom which
lets xenmgr issue powerbutton press shutdowns.

--- a/tools/libs/light/libxl_dm.c
+++ b/tools/libs/light/libxl_dm.c
@@ -1539,6 +1539,8 @@ static int libxl__build_device_model_arg
         }
         if (!libxl__acpi_defbool_val(b_info)) {
             flexarray_append(dm_args, "-no-acpi");
+        } else {
+            flexarray_append(dm_args, "-xen-acpi-pm");
         }
         if (b_info->max_vcpus > 1) {
             flexarray_append(dm_args, "-smp");
